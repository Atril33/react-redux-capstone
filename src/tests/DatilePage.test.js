import React from 'react';
import { useSelector } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { render, fireEvent, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Homepage from '../components/HomePage';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
}));

describe('<Homepage />', () => {
  const mockData = [
    { name: 'Test Country 1', flag: 'https://test-country-1-flag.png' },
    { name: 'Test Country 2', flag: 'https://test-country-2-flag.png' },
    { name: 'Test Country 3', flag: 'https://test-country-3-flag.png' },
  ];

  beforeEach(() => {
    useSelector.mockReturnValue({ countryData: mockData });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Should renders correctly', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <Homepage />
      </MemoryRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should displays all countries on load', () => {
    render(
      <MemoryRouter>
        <Homepage />
      </MemoryRouter>,
    );
    const countryImages = screen.getAllByRole('link', { name: /^test country/i });
    expect(countryImages).toHaveLength(mockData.length);
  });

  it('Should filters countries based on search', () => {
    render(
      <MemoryRouter>
        <Homepage />
      </MemoryRouter>,
    );

    const searchInput = screen.getByPlaceholderText('🔍 Search Country Here');
    fireEvent.change(searchInput, { target: { value: 'test country 2' } });

    const countryImages = screen.getAllByRole('link', { name: /^test country 2/i });
    expect(countryImages).toHaveLength(1);
  });
});
