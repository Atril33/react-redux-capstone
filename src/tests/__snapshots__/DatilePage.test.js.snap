// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Homepage /> Should renders correctly 1`] = `
Array [
  <div
    className="top-header"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="Back Icon"
        className="back-img"
        src="back.png"
      />
    </a>
    <h1>
      Countries
    </h1>
    <div
      className="right-icon"
    >
      <img
        alt="Microphone icon"
        src="microphone.png"
      />
      <img
        alt="Setting icon"
        className="setting-img"
        src="setting.png"
      />
    </div>
  </div>,
  <div
    className="search-cont"
  >
    <img
      alt="World Map"
      className="world-img"
      src="banar-img.jpg"
    />
  </div>,
  <input
    className="search-input"
    onChange={[Function]}
    placeholder="🔍 Search Country Here"
    type="text"
    value=""
  />,
  <div
    className="status-bar"
  >
    <h2
      className="country-status"
    >
      Stats by Country
    </h2>
  </div>,
  <div
    className="container"
  >
    <a
      className="small-container"
      href="/countries/Test Country 1"
      onClick={[Function]}
    >
      <img
        alt="Test Country 1"
        className="country-map"
        src="background.png"
      />
      <div
        className="country-info"
      >
        <h2
          className="country-name"
        >
          Test Country 1
        </h2>
        <h2
          className="country-population"
        />
      </div>
    </a>
    <a
      className="small-container"
      href="/countries/Test Country 2"
      onClick={[Function]}
    >
      <img
        alt="Test Country 2"
        className="country-map"
        src="background.png"
      />
      <div
        className="country-info"
      >
        <h2
          className="country-name"
        >
          Test Country 2
        </h2>
        <h2
          className="country-population"
        />
      </div>
    </a>
    <a
      className="small-container"
      href="/countries/Test Country 3"
      onClick={[Function]}
    >
      <img
        alt="Test Country 3"
        className="country-map"
        src="background.png"
      />
      <div
        className="country-info"
      >
        <h2
          className="country-name"
        >
          Test Country 3
        </h2>
        <h2
          className="country-population"
        />
      </div>
    </a>
  </div>,
]
`;
